engine/src/exec-interface-image.cpp

```diff
@@ -344,14 +352,39 @@ void MCImage::GetText(MCExecContext& ctxt, MCDataRef& r_text)
 		}
 
 		if (MCDataCreateWithBytes((const byte_t *)t_data, t_size, r_text))
-			return;
+        {
+            MCactiveimage = this;
+            return true;
+        }
 	}
 	
-	ctxt . Throw();
+	return false;
+}
+
+void MCImage::GetText(MCExecContext& ctxt, MCDataRef& r_text)
```

Comment:
Indeed, looking at the copy constructor - we should be able to do:
```
MCImage *t_image_to_operate_on = this;
if (m_rep != nullptr && m_rep->IsLocked())
{
   t_image_to_operate_on = new MCImage(*this);
}
t_image_to_operate_on->recompress();
...
if (t_image_to_operate_on != this)
{
  delete t_image_to_operate_on;
}
```
Note I suggest rewriting the original GetText() method to use t_image_to_operate_on rather than 'this'. The rationale being that this is a patch to fix a bug, but fixing the issue properly won't require this - so we want to keep all the patch code 'in the same place' as far as possible for later removal.